File: VB/MyLookUpEdit/MyLookUpEditStrategy.vb
Imports System.Collections.Generic
Imports DevExpress.Xpf.Grid.LookUp
Imports DevExpress.Data.Filtering

Namespace Q412037

    Friend Class MyLookUpEditStrategy
        Inherits LookUpEditStrategy

        Public Sub New(ByVal editor As LookUpEdit)
            MyBase.New(editor)
        End Sub

        Protected Overrides Function CreateDisplayFilterCriteria(ByVal text As String) As CriteriaOperator
            Dim lookUp As MyLookUpEdit = CType(Editor, MyLookUpEdit)
            Dim groupOperator As GroupOperator = New GroupOperator() With {.OperatorType = GroupOperatorType.Or}
            For Each col As ColumnItem In lookUp.ColumnsToFilter
                Dim list As List(Of CriteriaOperator) = New List(Of CriteriaOperator)()
                list.Add(New OperandProperty() With {.PropertyName = col.FieldName})
                list.Add(New OperandValue() With {.Value = text})
                groupOperator.Operands.Add(New FunctionOperator(FunctionOperatorType.Contains, list))
            Next

            Return groupOperator
        End Function
    End Class
End Namespace

File: CS/MyLookUpEdit/MyLookUpEditStrategy.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.Xpf.Grid.LookUp;
using DevExpress.Data.Filtering;
using DevExpress.Xpf.Grid;

namespace Q412037
{
    class MyLookUpEditStrategy : LookUpEditStrategy
    {
        public MyLookUpEditStrategy(LookUpEdit editor)
            : base(editor)
        {
        }
        protected override CriteriaOperator CreateDisplayFilterCriteria(string text)
        {
            MyLookUpEdit lookUp = (MyLookUpEdit)Editor;
            GroupOperator groupOperator = new GroupOperator() { OperatorType = GroupOperatorType.Or };
            foreach (ColumnItem col in lookUp.ColumnsToFilter)
            {
                List<CriteriaOperator> list = new List<CriteriaOperator>();
                list.Add(new OperandProperty() { PropertyName = col.FieldName });
                list.Add(new OperandValue() { Value = text });
                groupOperator.Operands.Add(new FunctionOperator(FunctionOperatorType.Contains, list));
            }
            return groupOperator;
        }
    }
}
